Orange/widgets/unsupervised/owdbscan.py

```diff
@@ -180,8 +182,11 @@ def set_data(self, data):
             self.Outputs.annotated_data.send(None)
             self.plot.clear_plot()
             return
-
-        if self.data is None:
+        if self.data.X is None or self.data.X.shape[1] == 0:
+            self.Error.no_features()
+            self.data = None
+            self.Outputs.annotated_data.send(None)
+            self.plot.clear_plot()
```

Comment:
To avoid the duplicated code, I'd rather the code was places right after the
```
        if not self.check_data_size(data):
            data = None
```
block, and implemented similarly (not necessarily with an additional function, but by setting the data to None before assigning it to `self.data`). In this case `self.data_normalized` would also be properly set.